core/src/main/java/apoc/trigger/TriggerMetadata.java

```diff
@@ -0,0 +1,196 @@
+    private final List<Relationship> createdRelationships;
+    private final List<Node> deletedNodes;
+    private final List<Relationship> deletedRelationships;
+    private final Map<String, List<Node>> removedLabels;
+    private final Map<String, List<PropertyEntryContainer<Node>>> removedNodeProperties;
+    private final Map<String, List<PropertyEntryContainer<Relationship>>> removedRelationshipProperties;
+    private final Map<String, List<Node>> assignedLabels;
+    private final Map<String, List<PropertyEntryContainer<Node>>> assignedNodeProperties;
+    private final Map<String, List<PropertyEntryContainer<Relationship>>> assignedRelationshipProperties;
+    private final Map<String, Object> metaData;
+
+    private TriggerMetadata(long transactionId, long commitTime,
+                            List<Node> createdNodes, List<Relationship> createdRelationships,
+                            List<Node> deletedNodes, List<Relationship> deletedRelationships,
+                            Map<String, List<Node>> removedLabels,
+                            Map<String, List<PropertyEntryContainer<Node>>> removedNodeProperties,
+                            Map<String, List<PropertyEntryContainer<Relationship>>> removedRelationshipProperties,
+                            Map<String, List<Node>> assignedLabels,
+                            Map<String, List<PropertyEntryContainer<Node>>> assignedNodeProperties,
+                            Map<String, List<PropertyEntryContainer<Relationship>>> assignedRelationshipProperties,
+                            Map<String, Object> metaData) {
+        this.transactionId = transactionId;
+        this.commitTime = commitTime;
+        this.createdNodes = createdNodes;
+        this.createdRelationships = createdRelationships;
+        this.deletedNodes = deletedNodes;
+        this.deletedRelationships = deletedRelationships;
+        this.removedLabels = removedLabels;
+        this.removedNodeProperties = removedNodeProperties;
+        this.removedRelationshipProperties = removedRelationshipProperties;
+        this.assignedLabels = assignedLabels;
+        this.assignedNodeProperties = assignedNodeProperties;
+        this.assignedRelationshipProperties = assignedRelationshipProperties;
+        this.metaData = metaData;
+    }
+
+    public static TriggerMetadata from(TransactionData txData) {
+        long txId, commitTime;
+        try {
+            txId = txData.getTransactionId();
+        } catch (Exception ignored) {
+            txId = -1L;
+        }
+        try {
+            commitTime = txData.getCommitTime();
+        } catch (Exception ignored) {
+            commitTime = -1L;
+        }
+        List<Node> createdNodes = Convert.convertToList(txData.createdNodes());
+        List<Relationship> createdRelationships = Convert.convertToList(txData.createdRelationships());
+        List<Node> deletedNodes = Convert.convertToList(txData.deletedNodes());
```

Comment:
I'm not confident that rebind on deleted entities actually works. Can we please ensure we have a test case that runs deletions and validates rebind on those workds?

Reply:
You're right, for these cases I converted them to Virtual Entities, so we can use them

Reply:
LGFM now, please feel free to merge. Thanks.